lib/gdal/dataset_mixins/algorithm_methods.rb

```diff
@@ -29,9 +29,9 @@ def rasterize_geometries!(band_numbers, geometries, burn_values,
         end
 
         gdal_options = GDAL::Options.pointer(options)
-        band_numbers = band_numbers.is_a?(Array) ? band_numbers : [band_numbers]
-        geometries = geometries.is_a?(Array) ? geometries : [geometries]
-        burn_values = burn_values.is_a?(Array) ? burn_values : [burn_values]
+        band_numbers = [band_numbers] unless band_numbers.is_a?(Array)
```

Comment:
You can use `Array(band_numbers)` as a shorthand for this, and it has the advantage of turning `nil` into `[]` instead of `[nil]`
```ruby
2.6.8 :001 > Array(1) #=> [1]
2.6.8 :002 > Array([1]) #=> [1]
2.6.8 :003 > Array(nil) #=> []
```